modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionsEvictManager.java

```diff
@@ -133,6 +151,81 @@ public void evictPartitionAsync(CacheGroupContext grp, GridDhtLocalPartition par
         }
     }
 
+    /**
+     *
+     * @param grp Cache group context.
+     * @param parts Partitions.
+     * @throws IgniteCheckedException If failed.
+     * @return Future.
+     */
+    public IgniteInternalFuture<Void> purgePartitionsExclusively(CacheGroupContext grp,
+        List<GridDhtLocalPartition> parts) throws IgniteCheckedException {
+        validateCacheGroupForExclusivePurge(grp);
+
+        if (F.isEmpty(parts))
+            return new GridFinishedFuture<>();
+
+        GroupEvictionContext grpEvictionCtx = evictionGroupsMap.computeIfAbsent(
+            grp.groupId(), (k) -> new GroupEvictionContext(grp));
+
+        ExclusivePurgeFuture fut = new ExclusivePurgeFuture(grpEvictionCtx, parts);
+
+        synchronized (mux) {
+            if (grpEvictionCtx.exclPurgeFut != null)
```

Comment:
1. We should add {} braces for each multi-line if branch.

2. How we expect callee handle the `IgniteCheckedException` we throw?